src/CloudWatchLogs/CloudWatchLogsClient.php

```diff
@@ -155,4 +156,46 @@
  * @method \Aws\Result updateLogAnomalyDetector(array $args = [])
  * @method \GuzzleHttp\Promise\Promise updateLogAnomalyDetectorAsync(array $args = [])
  */
-class CloudWatchLogsClient extends AwsClient {}
+class CloudWatchLogsClient extends AwsClient {
+    static $streamingCommands = [
+        'StartLiveTail' => true
+    ];
+
+    public function __construct(array $args)
+    {
+        parent::__construct($args);
+        $this->appendClientCustomMiddlewares();
+    }
+
+    private function appendClientCustomMiddlewares()
+    {
+        $this->getHandlerList()->appendInit($this->getSetStreamingFlagMiddleware(), 'streaming-flag-middleware');
+    }
+
+    private function getSetStreamingFlagMiddleware(): callable
+    {
+        return function (callable $handler) {
+            return function (CommandInterface $command, $request = null) use ($handler) {
+                if (!empty(self::$streamingCommands[$command->getName()])) {
+                    $command['@http']['stream'] = true;
+                }
+
+                return $handler($command, $request);
+            };
+        };
+    }
+
+    public function startLiveTailCheckingForResults($args): \Generator
```

Comment:
Where is this used?

Reply:
This is just a helper method on top of startLiveTail command that yields the results when they are available.